include/deal.II/fe/fe_values.h

```diff
@@ -2708,6 +2708,46 @@ class FEValuesBase :
    */
   UpdateFlags compute_update_flags (const UpdateFlags update_flags) const;
 
+  struct MappingDescription
```

Comment:
I'm not a fan of this. You're trying to move decisions that are germane only to the mapping classes to `FEValues`, but I don't think that that's the place for it. Leave it to the mappings to decide what exactly they want to do with `cell_similarity`.